tools/gfx/metal/metal-device.cpp

```diff
@@ -316,29 +321,29 @@ Result DeviceImpl::createTextureResource(
         break;
     }
 
-    NS::UInteger arrayLength = calcEffectiveArraySize(desc);
+    bool isArray = desc.arraySize > 0;
```

Comment:
Shouldn't this be 1 instead of 0?
```
bool isArray = desc.arraySize > 1;
```
The previous code was using `> 1`, and it would make more sense to be an array when the size is bigger than 1.

Reply:
There is a difference between an 1-layer array texture vs a normal texture. To distinguish between them, we use size=0 to mean not an array, and 1 to mean a size-1 array.

Reply:
exactly